wolcomm/generic-ip-rs
# prefix sets

- initial commit
- ignore profiling artifacts
- skip glue nodes during iteration
- de-dup bench functions
- only allocate once per add
- remove redunant IpPrefix enum
- formatting
- optimize prefix structs
- rename prefix trait
- first correct implementation
- add ci/cd config
- publish tags only
- execute test suites in parallel
- fmt sources
- don't try to bench on gh-actions
- fix lint errors
- don't require benchmarks for publish job
- format utils
- set api cleanup
- remove duplication in sub-prefix iterators
- don't store ipnet in ipprefix
- remove unused impls
- avoid unnecessary matches
- refactor node implementation
- remove unused test
- additional set methods
- eliminate multiple tree traversals for de-dup/aggregate/cleanup
- rename IpPrefix::BitMap to ::Bits
- impl from IpPrefix for IpPrefixRange
- refactor iterator names
- implement partial ordering on prefixes
- more basic trait impls
- test for send/sync regressions
- custom error types
- add docs
- add cargo metadata and version sync tests
- add license
- update README and fix some typos
- add 'versions' test suite to ci
- fix test job
- don't generate coverage for doctests
- add badges
- set rc version
- fix badge links
- correct name in README
- fix urls in badges
- add ranges example
- add bgpq3 aggregation comparison
- add TryFrom<IpNet> impls
- work in progress
- refactor to use `generic-ip` types
- prepare repo merge
- refactor `prefixset` into `ip::concrete::prefix::set`
- make `clippy` happy
- fix search direction bug
- don't be verbose
- tidy up prefix-sets bench
- don't test benchmarks
- check docs generation in CI
- fix typo in job name
- fix imports in doctests
- fix broken links in docs
- exclude `prefix-sets` integration tests from CI
- fix documentation typos
- split `traits::prefix` into sub-modules
- add `PrefixSet` trait and `any::PrefixSet`
- fix fuzz targets
- fix new `clippy` lints
